src/main/ruby/truffleruby/core/random.rb

```diff
@@ -66,10 +66,24 @@ def random(limit)
   end
 
   def random_range(limit)
-    min, max = limit.max.coerce(limit.min)
+    return nil if Comparable.compare_int(limit.end <=> limit.begin) < 0
+
+    min, max = limit.end.coerce(limit.min)
+    max = if limit.exclude_end?
```

Comment:
It should be the other way around: random() below always uses exclusive end semantics.
So if `exclude_end?`, just use `max`, and if `exclude_end? == false` then `.next_float` for Float and `.succ` otherwise (also works for Integer).

Reply:
yep, I was confused in the beginning what you were referring to, now I see that ```random_*```` methods are all exclusive